pkg/kubelet/prober/prober.go

```diff
@@ -219,16 +219,25 @@ type execInContainer struct {
 	run func() ([]byte, error)
 }
 
-func (p *prober) newExecInContainer(container api.Container, containerID kubecontainer.ContainerID, cmd []string) exec.Cmd {
+func (p *prober) newExecInContainer(container api.Container, containerID kubecontainer.ContainerID, cmd []string, timeout time.Duration) exec.Cmd {
 	return execInContainer{func() ([]byte, error) {
 		var buffer bytes.Buffer
 		output := ioutils.WriteCloserWrapper(&buffer)
-		err := p.runner.ExecInContainer(containerID, cmd, nil, output, output, false)
-		if err != nil {
-			return nil, err
+		errCh := make(chan error, 1)
+		go func() {
+			glog.V(4).Infof("Executing readiness probe using : %#v", cmd)
+			errCh <- p.runner.ExecInContainer(containerID, cmd, nil, output, output, false)
+		}()
+		select {
+		case err := <-errCh:
+			glog.V(4).Infof("Readiness probe finished %v - %v", errCh, string(buffer.Bytes()))
+			if err != nil {
+				return nil, err
+			}
+			return buffer.Bytes(), nil
+		case <-time.After(timeout):
```

Comment:
This is inconsistent with the other 2 probes that use an `http.Client`.
_A Timeout of zero means no timeout._